<?xml version="1.0"?>
<launch>
  <!--<node name="check_ueye_api" pkg="ueye_cam" type="check_ueye_api" required="true" /> -->
  <arg name="nodelet_manager_name" default="nodelet_manager" />
  <arg name="camera_name" default="ueye_enac_ceiling_1_6mm" />
  <arg name="camera_id" default="128" />
  <arg name="cam_nodelet_id" default="1" />
  
  <node pkg="nodelet" type="nodelet" name="ueye_cam_nodelet_$(arg cam_nodelet_id)"
        args="load ueye_cam/ueye_cam_nodelet $(arg nodelet_manager_name)">
    <param name="camera_name" type="str" value="$(arg camera_name)" />
    <param name="camera_topic" type="str" value="image_raw" />
    <param name="camera_id" type="int" value="$(arg camera_id)" />
    <param name="camera_intrinsics_file" type="string" value="" />
    <param name="camera_parameters_file" type="string" value="" />
    <param name="ext_trigger_mode" type="bool" value="False" /> 

    <param name="color_mode" type="str" value="mono8" />

    <param name="image_width" type="int" value="1600" />
    <param name="image_height" type="int" value="1200" />

    <param name="auto_gain" type="bool" value="True" />

    <param name="auto_exposure" type="bool" value="False" />
    <param name="exposure" type="int" value="8" /><!-- in ms -->

    <param name="frame_rate" type="double" value="30.41" /> <!-- 60 -->

    <param name="pixel_clock" type="int" value="64" /> <!-- 128 -->
  </node>

</launch>
